app/jobs/regular/auto_join_channel_batch.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Jobs
+  class AutoJoinChannelBatch < ::Jobs::Base
+    def execute(args)
+      return "starts_at or ends_at missing" if args[:starts_at].blank? || args[:ends_at].blank?
+      return "End is higher than start" if args[:ends_at] < args[:starts_at]
+
+      channel = ChatChannel.find_by(id: args[:chat_channel_id])
```

Comment:
We can probably just do `ChatChannel.find(...)`  and remove next line, given we want to fail on not found

Reply:
Or maybe even better: `ChatChannel.find_by!(id: ..., chatable_type: 'Category', auto_join_users: true)`   ?

That way we can remove all the checks after

Reply:
`find_by!` will return super early, if you mean you just don't want this to create a not found error, you can just do `find_by` and then `return if !channel`

Reply:
Yeah, I meant rising an error `find_by` + `return if !channel` is the way to go IMO.